File: Python/5.py
def main():
    print(wholeLCM([i for i in range(1,21)]))

def wholeLCM(A):
    if len(A)==1:
        return A[0]
    else:
        return lcm(A[0], wholeLCM(A[1:]))

def gcd(a,b):
    """Compute the greatest common divisor of a and b"""
    while b > 0:
        a, b = b, a % b
    return a
    
def lcm(a, b):
    """Compute the lowest common multiple of a and b"""
    return a * b / gcd(a, b)


if __name__=="__main__":
    main()

File: Rust/5.rs
fn main() {
    let mut move_lcm = 1;
    for i in 1..21 {
        move_lcm = lcm(move_lcm, i);
    }
    println!("Result: {}", move_lcm);
}

fn lcm(a: u64, b: u64) -> u64 {
    a*b / gcd(a,b)
}

fn gcd(a: u64, b: u64) -> u64 {
    if b == 0 { a } else { gcd(b, a%b) }
}
